packages/leancode_flutter_svg_adaptive_loader/lib/src/leancode_flutter_svg_adaptive_loader.dart

```diff
@@ -0,0 +1,149 @@
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:vector_graphics/vector_graphics.dart';
+import 'package:vector_graphics_compiler/vector_graphics_compiler.dart' as vg;
+
+/// A [BytesLoader] able to distinguish xml-based svg from compiled binary form,
+/// by looking for '<svg' pattern at 1000 of first characters.
+/// Using logic from [AssetBytesLoader] for binaries and [SvgAssetLoader]
+/// for standard xml-based format.
+class LeancodeFlutterSvgAdaptiveLoader extends BytesLoader {
+  /// Creates a [BytesLoader]
+  const LeancodeFlutterSvgAdaptiveLoader(
+    this.assetName, {
+    this.packageName,
+    this.assetBundle,
+  });
+
+  /// The name of the asset to load.
+  final String assetName;
+
+  /// The package name to load from, if any.
+  final String? packageName;
+
+  /// The asset bundle to use.
+  ///
+  /// If unspecified, [DefaultAssetBundle.of] the current context will be used.
+  final AssetBundle? assetBundle;
+
+  AssetBundle _resolveBundle(BuildContext? context) {
+    if (assetBundle != null) {
+      return assetBundle!;
+    }
+    if (context != null) {
+      return DefaultAssetBundle.of(context);
+    }
+    return rootBundle;
+  }
+
+  Future<ByteData> _prepareMessage(BuildContext? context) {
+    return _resolveBundle(context).load(
+      packageName == null ? assetName : 'packages/$packageName/$assetName',
+    );
+  }
+
+  String _decode(ByteData? message, [int? lenght]) =>
+      utf8.decode(message!.buffer.asUint8List(0, lenght), allowMalformed: true);
```

Comment:
Why are we asserting here?

Reply:
That's how the author coded it in the `flutter_svg` package.